batch/test/test_dag.py

```diff
@@ -226,7 +226,7 @@ def test_input_dependency_directory(client):
                             parent_ids=[head.id])
     batch.close()
     tail.wait()
-    assert head.status()['exit_code']['main'] == 0, head.cached_status()
+    assert head.status()['exit_code']['main'] == 0, head._status
```

Comment:
Should be head.status(); you're returning a Batch, which doesn't have a _status property, (only aioclient Job has this property).

Reply:
It's a Job. I don't want to call status again because that could cause it to change. I'll add _status to the synchronous job though